lib/instrumentation/async-hooks.js

```diff
@@ -3,10 +3,19 @@
 const asyncHooks = require('async_hooks')
 const shimmer = require('./shimmer')
 
+// FOR INTERNAL TESTING PURPOSES ONLY!
+const resettable = process.env._ELASTIC_APM_ASYNC_HOOKS_RESETTABLE === 'true'
+let _asyncHook
+
 module.exports = function (ins) {
   const asyncHook = asyncHooks.createHook({ init, before, destroy })
   const contexts = new WeakMap()
 
+  if (resettable) {
+    if (_asyncHook) _asyncHook.disable()
+    _asyncHook = asyncHook
+  }
```

Comment:
Is there a reason not to just _always_ do this? 🤔

Reply:
Good question... I did so in the first iteration but I wasn't 100% sure if there were any implications of keeping a reference to this object so it couldn't be garbage collected. This way it only kept it around during testing - but it's probably irellevant 🤷‍♂

Reply:
Fair. Approved anyway just to get the fix in. I can clean it up later. 🤔